biorbd_optim/optimal_control_program.py

```diff
@@ -516,19 +516,33 @@ def solve(self, solver="ipopt", show_online_optim=False, options_ipopt={}):
         # Solve the problem
         return solver.call(arg)
 
-    def save(self, sol, file_path, to_numpy=False, **parameters):
+    def save(self, sol, file_path):
         _, ext = os.path.splitext(file_path)
         if ext == "":
             file_path = file_path + ".bo"
+        elif ext != ".bo":
+            raise RuntimeError(f"Incorrect extension({ext}), it should be (.bo) or (.bob) if you use save_get_data.")
+
+        OptimalControlProgram._save_with_pickle(
+            {"ocp_initilializer": self.original_values, "sol": sol, "versions": self.version}, file_path
+        )
+
+    def save_get_data(self, sol, file_path, **parameters):
+        _, ext = os.path.splitext(file_path)
+        if ext == "":
+            file_path = file_path + ".bob"
+        elif ext != ".bob":
+            raise RuntimeError(f"Incorrect extension({ext}), it should be (.bob) or (.bo) if you use save.")
+
+        OptimalControlProgram._save_with_pickle({"data": Data.get_data(self, sol["x"], **parameters)}, file_path)
```

Comment:
Last thing, since the goad is to have it in numpy (and not DM) maybe we should go through the dict before pickle it and convert the DM in numpy?

Reply:
numpy contains just the result of get_data in order to be able to exploit the data via Eigen, without any call to Casadi, and as ocp is very much linked to Casadi, it is easier to directly save the result of get_data with Casadi.